src/management/api/analytics/logs/log.component.ts

```diff
@@ -36,6 +38,42 @@ const LogComponent: ng.IComponentOptions = {
 
       return null;
     };
+    this.getCurl = function(log, scope){
+      var request = (scope == "client") ? log.clientRequest : log.proxyRequest;
+      if(scope == "client"){
+        var uri = request.headers['X-Forwarded-Proto']+"://"+request.headers['Host']+request.headers['X-Forwarded-Prefix']+request.uri;
```

Comment:
`request.headers['X-Forwarded-Prefix']` is not always provided, you should take care about it and check if the header is existing or not.

Reply:
The same rule must be applied for `request.headers['X-Forwarded-Proto']`

Reply:
Hi @brasseld you're right i forgot that point, but i'm not sur of how to handle that...
```suggestion
        var proto = (request.headers['X-Forwarded-Proto']) ? request.headers['X-Forwarded-Proto']+"://" : "";
        var prefix = (request.headers['X-Forwarded-Prefix']) ? request.headers['X-Forwarded-Prefix'] : "";
        var uri =proto+request.headers['Host']+prefix+request.uri;
```
Would it be ok ?

Reply:
I'm not sure about `proto` part. IMHO, there is no way currently to get the scheme of the request if you're not relying on the `X-Forwarded-Proto`. Meaning that there is probably something to do in Gravitee to take care of the proto and save it as part of the request's log.

The other point is about the `Host` header. If you're relying on `X-Forwarded-Proto`, I suppose you would also have to extract the value of `X-Forwarded-Host` if this later is provided, no?

Ok, for the path-prefix.

Reply:
I'm totally agree with you...
The best solution is to have all of these informations without reading the headers.
I would have loved helping for the gravitee back part, but I don't feel comfortable with Java (yet)...

Reply:
No worries, I'm understanding quite perfectly your position... and I know I have to decline all Java code you may provide in the future :-P

Let me see what can we do to manage proto and host.